frontend/server/src/Controllers/User.php

```diff
@@ -3913,20 +3913,27 @@ public static function isNonUserIdentity(
      */
     public static function getLoginDetailsForSmarty(\OmegaUp\Request $r) {
         $emailVerified = true;
-        $redirectResponse = null;
-        $thirdPartyLogin = $r->ensureString('third_party_login');
-
-        if ($thirdPartyLogin === 'linkedin') {
-            $code = $r->ensureString('code');
-            $state = $r->ensureString('state');
-            $redirect = $r->ensureString('redirect');
-            \OmegaUp\Controllers\Session::loginViaLinkedIn(
-                $code,
-                $state,
-                $redirect
-            );
-        } elseif ($thirdPartyLogin === 'facebook') {
-            \OmegaUp\Controllers\Session::loginViaFacebook();
+        $thirdPartyLogin = $r->ensureOptionalString('third_party_login') ?? '';
```

Comment:
```suggestion
        $thirdPartyLogin = $r->ensureOptionalString('third_party_login');
```

siempre que sea posible hay que evitar el uso de valores mágicos que no son valores válidos: `''` no es un valor válido para `third_party_login`, y `null` es un valor perfectamente razonable para esto: indica que no está definido.

Reply:
Listo